tests/BootstrapperTest.php

```diff
@@ -380,3 +381,30 @@ function getDiskPrefix(string $disk): string
 
     return $prefix;
 }
+
+test('url bootstrapper overrides the root url when tenancy gets initialized and reverts the url to the central one after tenancy ends', function() {
+    config(['tenancy.bootstrappers.url' => UrlTenancyBootstrapper::class]);
+
+    UrlTenancyBootstrapper::$rootUrlOverride = function (Tenant $tenant) {
+        $baseUrl = URL::to('/');
+        $scheme = str($baseUrl)->before('://') . '://';
+
+        return str($baseUrl)
+            ->after($scheme)
+            ->prepend($tenant->getTenantKey() . '.')
+            ->prepend($scheme)
+            ->toString();
+     };
+
+     $tenant = Tenant::create();
+
+     expect(URL::to('/'))->not()->toContain($tenant->getTenantKey());
+
+     tenancy()->initialize($tenant);
+
+     expect(URL::to('/'))->toContain($tenant->getTenantKey());
+
+     tenancy()->end();
+
+     expect(URL::to('/'))->not()->toContain($tenant->getTenantKey());
```

Comment:
Maybe add direct assertions about the routes, so that we know that the generated URLs don't have e.g. duplicate `/`s etc. Rather than just "contains" assertions.

Reply:
I updated the test. I also thought it was a good idea to check if `url(route())` generates the URL correctly. Could you take a look at [the commit](https://github.com/archtechx/tenancy/pull/1044/commits/3fbf6b4a45932a0d4ccbae3c363d2198fa9e4648) and let me know what you think now?

Reply:
I was thinking about using literal strings (`http://tenant1.localhost/foo`), but checking that it equals to the value returned from the closure is probably good enough since it checks directly that the closure is being used 👍🏻